### PR TITLE
feat!: support static gas price setting in `NodeConfig`

### PR DESCRIPTION
**Pull request by MujkicA**
Originally, this PR was aiming to solve several bugs and enables a static gas price in our tests so that we can catch said bugs.
It was split into several smaller PRs:
#1423, #1425, #1426

In it's current version, it only adds the static gas price config parameter to the node config and enables it in our tests.

BREAKING CHANGE: Adds the field `static_gas_price` to `NodeConfig`

### Checklist
- [ ] I have linked to any relevant issues.
- [ ] I have updated the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary labels.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by Br1ght0ne — DISMISSED** [review]

File: examples/cookbook/src/lib.rs [hunk 1]
```diff
@@ -190,13 +191,18 @@ mod tests {
 
         let tx = tb.build(&provider).await?;
 
+        dbg!(&tx);
```
Comment (hal3e) [inline examples/cookbook/src/lib.rs:194]:
```suggestion
```

File: packages/fuels-core/src/types/transaction_builders.rs [hunk 2]
```diff
@@ -554,6 +554,7 @@ impl ScriptTransactionBuilder {
             0
         } else {
             let dry_run = if let Some(dry_run) = dry_runner.last_dry_run() {
+                // TODO: ADD THIS TO DOCS
```
Comment (hal3e) [inline packages/fuels-core/src/types/transaction_builders.rs:557]: can you please create an issue for this?

**Review by hal3e — CHANGES_REQUESTED** [review]
Looks very good. Have left some nits

**Comment by hal3e** [comment]
can you please add the `BREAKING CHANGE` section in the PR description.

**Review by hal3e — APPROVED** [review]
awesome work! thanks

**Review by Br1ght0ne — APPROVED** [review]

**Review by digorithm — APPROVED** [review]

**Pull request merged**